hawc_hal/maptree/from_hdf5_file.py

```diff
@@ -66,6 +68,19 @@ def from_hdf5_file(map_tree_file, roi):
 
     data_analysis_bins = collections.OrderedDict()
 
+    # Figure out the transits
+    transits_bins = []
+    for bin_name in bin_names:
+        this_meta = meta_df.loc[bin_name]
+        # Specify n_transits (or not), default value contained in map is this_meta['n_transits']
+        transits_bins.append(this_meta['n_transits'])
+
+    # pick out the transits same as root file
+    use_transits = np.max(transits_bins)
+    if n_transits!=None:
```

Comment:
This is not mandatory, but for easier readability, I suggest to put the current lines 72 to 76 also into the if block, so that it's easier to see which code is used in what circumstance. Also, there's a much easier way to get the max of some elements in a dataframe. My suggestion:

```
if n_transits is None:
    use_transits = meta_df.iloc[bin_names]["n_transits"].max()
else:
    use_transits = n_transits
```

Or, since `bin_names` always seems to contain all bins (not the user-specified ones), `meta_df["n_transits"].max() ` should work instead and be even faster.

Reply:
I with with your second suggestion `meta_df["n_transits"].max()`, as it seemed much cleaner to me.